pkg/golinters/gocritic/gocritic.go

```diff
@@ -494,49 +495,49 @@ func (s *goCriticSettingsWrapper) validateOptionsCombinations() error {
 	return nil
 }
 
-func (s *goCriticSettingsWrapper) validateCheckerTags() error {
+func (s *settingsWrapper) validateCheckerTags() error {
 	for _, tag := range s.EnabledTags {
 		if !s.allChecksByTag.has(tag) {
-			return fmt.Errorf("enabled tag %q doesn't exist, see %s's documentation", tag, goCriticName)
+			return fmt.Errorf("enabled tag %q doesn't exist, see %s's documentation", tag, name)
 		}
 	}
 
 	for _, tag := range s.DisabledTags {
 		if !s.allChecksByTag.has(tag) {
-			return fmt.Errorf("disabled tag %q doesn't exist, see %s's documentation", tag, goCriticName)
+			return fmt.Errorf("disabled tag %q doesn't exist, see %s's documentation", tag, name)
 		}
 	}
 
 	return nil
 }
 
-func (s *goCriticSettingsWrapper) validateCheckerNames() error {
+func (s *settingsWrapper) validateCheckerNames() error {
 	for _, name := range s.EnabledChecks {
 		if !s.allChecks.has(name) {
-			return fmt.Errorf("enabled check %q doesn't exist, see %s's documentation", name, goCriticName)
+			return fmt.Errorf("enabled check %q doesn't exist, see %s's documentation", name, name)
 		}
 	}
 
 	for _, name := range s.DisabledChecks {
 		if !s.allChecks.has(name) {
-			return fmt.Errorf("disabled check %q doesn't exist, see %s documentation", name, goCriticName)
+			return fmt.Errorf("disabled check %q doesn't exist, see %s documentation", name, name)
 		}
 	}
 
 	for name := range s.SettingsPerCheck {
 		lcName := strings.ToLower(name)
 		if !s.allChecksLowerCased.has(lcName) {
-			return fmt.Errorf("invalid check settings: check %q doesn't exist, see %s documentation", name, goCriticName)
+			return fmt.Errorf("invalid check settings: check %q doesn't exist, see %s documentation", name, name)
 		}
 		if !s.inferredEnabledChecksLowerCased.has(lcName) {
-			s.logger.Warnf("%s: settings were provided for disabled check %q", goCriticName, name)
+			s.logger.Warnf("%s: settings were provided for disabled check %q", name, name)
```

Comment:
`name` is shadowed. Looks like we should use a different variable name or leave the `goCriticName`.